eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Utxow.hs

```diff
@@ -124,20 +122,11 @@ data ConwayUtxowPredFailure era
   | MissingRedeemers
       ![(PlutusPurpose AsItem era, ScriptHash (EraCrypto era))]
   | MissingRequiredDatums
-      -- | Set of missing data hashes
-      !(Set (DataHash (EraCrypto era)))
-      -- | Set of received data hashes
-      !(Set (DataHash (EraCrypto era)))
+      !(Mismatch 'RelEQ (Set (DataHash (EraCrypto era)))) -- These values are serialised in reverse order
```

Comment:
This is not a `Mismatch`. This predicate failure specifies which datums are missing. In fact we'll need to change this predicate failure to:

```haskell
  | MissingRequiredDatums
      -- | Set of missing data hashes
      !(NonEmpty (DataHash (EraCrypto era)))
      -- | Set of received data hashes
      !(Set (DataHash (EraCrypto era)))
```

So, please revert all changes to this predicate failure

Reply:
We even have a ticket about it: https://github.com/IntersectMBO/cardano-ledger/issues/4066

Reply:
I have added this as a comment 👍 Thanks!